assets/js/src/settings/store/actions.ts

```diff
@@ -0,0 +1,15 @@
+import { Action, Settings } from './types';
+
+export function setSetting(path: string[], value: any): Action {
+  return { type: 'SET_SETTING', path, value };
+}
+
+export function* saveSettings(data: Settings) {
+  yield { type: 'SAVE_STARTED' };
+  const error = yield { type: 'SEND_DATA_TO_API', data };
+  if (error) {
+    return { type: 'SAVE_FAILED', error };
+  }
+  yield { type: 'TRACK_SETTINGS_SAVED', data };
+  return { type: 'SAVE_DONE' };
+}
```

Comment:
Just a note: In the form editor, we decided to keep actions without any logic and we [put the flow into control](https://github.com/mailpoet/mailpoet/blob/master/assets/js/src/form_editor/store/controls.jsx#L36). The idea behind this decision is to have the logic in one place. It is not a best practice or something we need to enforce.
We also don't pass all the data we want to save to the action but we [select it from the store within the code of the control.](https://github.com/mailpoet/mailpoet/blob/master/assets/js/src/form_editor/store/controls.jsx#L45) That way you don't have to distribute those data in every component which could possibly trigger saving.